### DISCUSSION TITLE
Manage old exist magento projects

### DISCUSSION THREAD

**Discussion by alkrenawizid** [markshust/docker-magento#715] [Q&A]
Hi @markshust,
**1. Is it possible to manage exist old Magento projects with this docker or it's only for development?**
**2. What are the options to manage the below environments until I upgrade them to Magento 2.4.4 with this docker?** 
Currently my clients have old Magento project (Magento ver. 2.3.0, 2.3.5-p1 & 2.3.5-p2),

I see that you recently add support for php 7.4 and I already try to install Magento ver. 2.3.5-p1 via Automated Setup, 
But get php compatibility error:

```
 Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires magento/product-community-edition 2.3.5-p1 -> satisfiable by magento/product-community-edition[2.3.5-p1].
    - magento/product-community-edition 2.3.5-p1 requires php ~7.1.3||~7.2.0||~7.3.0 -> your php version (7.4.29) does not satisfy that requirement.
  Problem 2
    - Root composer.json requires dealerdirect/phpcodesniffer-composer-installer ^0.5.0 -> satisfiable by dealerdirect/phpcodesniffer-composer-installer[v0.5.0].
    - dealerdirect/phpcodesniffer-composer-installer v0.5.0 requires composer-plugin-api ^1.0 -> found composer-plugin-api[2.2.0] but it does not match the constraint.
  Problem 3
    - Root composer.json requires magento/magento2-functional-testing-framework ~2.6.4 -> satisfiable by magento/magento2-functional-testing-framework[2.6.4, 2.6.5, 2.6.6, 2.6.7].
    - magento/magento2-functional-testing-framework[2.6.4, ..., 2.6.7] require php ~7.1.0||~7.2.0||~7.3.0 -> your php version (7.4.29) does not satisfy that requirement.
```

Thanks,
Zid.

**Comment by markshust** [comment]
I'd be open to taking in some sort of documentation that links the versions of docker-magento to versions of Magento/PHP. I always recommend version controlling your setup right after install, so you can always peg that project setup to an appropriate Magento version. But I do know there are times that you may take in or assume the responsibilities of a legacy project, in which case I don't have a great answer for you right now.

There are versions of docker-magento that run on pretty much every version of Magento from version 2.3.0 onwards. The problem is identifying the project tag at that version. It's possible you can deduce this by the times these releases were made with the time that Magento was around when that version came out.

This is a common problem/question, and I'd love to add some sort of documentation, custom script, etc. that can switch everything over to that version. Unfortunately, I do not have such time for an undertaking, so I'll leave it up to the community to build something out for this if they wish 😁 and I'd gladly take in a PR that gets us in a step to this direction.
